### PR TITLE
Refactor neutronics pka generators

### PR DESCRIPTION
**Pull request by snschune**
Refactor neutronics pka generators to include averaged number densites as input. Add documentation.

### CODE REVIEW

**Comment by snschune** [comment]
Wait not done yet!

**Comment by snschune** [comment]
To be completed:

- [ ] documentation for PKAGeneratorRecoil
- [ ] test for  PKAGeneratorRecoil

File: doc/content/documentation/systems/UserObjects/MyTRIMRasterizer.md [hunk 1]
```diff
@@ -1,14 +1,68 @@
...
-the typical documentation associated with a MooseObject; however, what is contained is ultimately
-determined by what is necessary to make the documentation clear for users.
+## Providing isotopic species and concentrations
+
+`MyTRIMRasterizer` uses information of the isotopic composition in the domain
+provided by MooseVariables specified in parameter `var` to prepare a rasterized
+domain on which binary collision Monte Carlo is performed. Rasterization is the
+process of averaging smoothly varying compositional data over each FEM mesh element.
+
+The entries in of the `var`, `M`, and `Z` parameters are paired; we index them as $\gamma_j(\vec{r})$, $M_j$, and $Z_j$, respectively.
```
Comment (dschwen) [inline doc/content/documentation/systems/UserObjects/MyTRIMRasterizer.md]: _in of_?

File: doc/content/index.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 <h2 class="center" style="font-size:150%;font-weight:300;">Mesoscale-Atomistics Glue Program for Integrated Execution</h2>
 
 Welcome to the Magpie project. Magpie is an Idaho National Laboratory tool for coupling
-atomistic and Monte Carlo methods to the finite element MOOSE framework. Iy currently couples
-the MyTRIM binary collision Montecarlo code and teh Sandia SPPARKS Monte Carlo toolkit.
+atomistic and Monte Carlo methods to the finite element MOOSE framework. It currently couples
+the MyTRIM binary collision Monte-Carlo code and the Sandia SPPARKS Monte Carlo toolkit.
```
Comment (dschwen) [inline doc/content/index.md]: 👍

File: doc/hidden.yml [hunk 3]
```diff
@@ -21,7 +21,6 @@
 - /UserObjects/NeutronicsSpectrumSamplerFission
 - /UserObjects/PKAConstant
 - /UserObjects/PKAFissionFragmentEmpirical
-- /UserObjects/PKAFissionFragmentNeutronics
```
Comment (dschwen) [inline doc/hidden.yml]: Unhide `/UserObjects/MyTRIMRasterizer` as well

**Comment by dschwen** [comment]
@snschune unfortunately the MooseDocs update I just merged will require some modification of this PR. Let me see if I can figure out an easy way to do this...

**Comment by dschwen** [comment]
Ok, I manually merged this.

**Pull request closed without merging**